app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java

```diff
@@ -24,12 +24,34 @@
 import net.gsantner.markor.ui.hleditor.TextActions;
 import net.gsantner.opoc.util.Callback;
 import net.gsantner.opoc.util.ContextUtils;
+import net.gsantner.opoc.util.StringUtils;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.regex.Pattern;
 
 public class MarkdownTextActions extends TextActions {
 
+    private static final Pattern PREFIX_ORDERED_LIST = Pattern.compile("^(\\s*)(\\d+\\.\\s)");
+    private static final Pattern PREFIX_ATX_HEADING = Pattern.compile("^(\\s{0,3}#{1,6}\\s)");
+    private static final Pattern PREFIX_QUOTE = Pattern.compile("^(>\\s)");
+    private static final Pattern PREFIX_CHECKED_LIST = Pattern.compile("^(\\s*)((:?-|\\*)\\s\\[(:?x|X)]\\s)");
+    private static final Pattern PREFIX_UNCHECKED_LIST = Pattern.compile("^(\\s*)((:?-|\\*)\\s\\[\\s]\\s)");
+    private static final Pattern PREFIX_UNORDERED_LIST = Pattern.compile("^(\\s*)((:?-|\\*)\\s)");
+    private static final Pattern PREFIX_LEADING_SPACE = Pattern.compile("^(\\s*)");
+
+    private static final Pattern[] PREFIX_PATTERNS = {
+            PREFIX_ORDERED_LIST,
+            PREFIX_ATX_HEADING,
+            PREFIX_QUOTE,
+            PREFIX_CHECKED_LIST,
+            PREFIX_UNCHECKED_LIST,
+            // Unordered has to be after checked list
+            PREFIX_UNORDERED_LIST,
+            PREFIX_LEADING_SPACE,
+    };
+
```

Comment:
should these patterns go somewhere else?

Reply:
do we have similar in plaintext format as well?

Reply:
We do not have similar patterns in plain text. I have made changes only to markdown behavior.

Reply:
hm then it was something different :smile: